core/jvm/src/main/scala/fs2/compression.scala

```diff
@@ -243,14 +243,17 @@ object compression {
       deflateParams: DeflateParams,
       deflater: Deflater,
       crc32: Option[CRC32]
-  ): Pipe[F, Byte, Byte] =
-    _deflate_stream(deflateParams, deflater, crc32)(_).stream
+  ): Pipe[F, Byte, Byte] = {
+    val deflatedBuffer = new Array[Byte](deflateParams.bufferSizeOrMinimum)
```

Comment:
This change makes the returned pipe unsafe for reuse -- consider calling this method once but then using resulting pipe in 2 or more streams.

Reply:
Good catch, and suggestion, thanks